src/mesh/forest/logical_location.cpp

```diff
@@ -118,10 +118,10 @@ bool LogicalLocation::IsNeighbor(const LogicalLocation &in) const {
 
   bool neighbors = true;
   for (int dir = 0; dir < 3; ++dir) {
-    auto low = (l(dir) << level_shift_this) - 1;
+    auto low = l(dir) * block_size_this - 1;
     auto hi = low + block_size_this + 1;
 
-    auto low_in = (in.l(dir) << level_shift_in);
+    auto low_in = in.l(dir) * block_size_in;
     auto hi_in = low_in + block_size_in - 1;
```

Comment:
Out of curiosity, where does the asymmetry come from (i.e., that low/hi is offset by-1/+1 and low_in, hi_in by 0,-1)?
Would be great to also add a short comment in the code.

Reply:
It just has to do with how the overlap is checked, i.e. the bounds of `this` are increased by one in the positive and negative direction to create a one layer thick ghost region.